contrib/codeanalysis/src/python/pants/contrib/codeanalysis/tasks/index_java.py

```diff
@@ -74,5 +81,22 @@ def entries_file(_vt):
           raise TaskError('java {main} ... exited non-zero ({result})'.format(
             main=self._KYTHE_INDEXER_MAIN, result=result))
 
+    dist = self.get_options().dist
     for vt in invalidation_check.all_vts:
-      self.context.products.get_data('kythe_entries_files', dict)[vt.target] = entries_file(vt)
+      entries = entries_file(vt)
+      self.context.products.get_data('kythe_entries_files', dict)[vt.target] = entries
+      if dist != 'none':
+        kythe_distdir = os.path.join(self.get_options().pants_distdir, 'kythe')
+        safe_mkdir(kythe_distdir)
+        uncompressed_kythe_distpath = os.path.join(
+          kythe_distdir, '{}.entries'.format(vt.target.address.path_safe_spec))
+        if dist == 'uncompressed':
+          kythe_distpath = uncompressed_kythe_distpath
+          shutil.copy(entries, kythe_distpath)
+        else:
+          kythe_distpath = shutil.make_archive(base_name=uncompressed_kythe_distpath,
```

Comment:
I don't think it really matters, but: we have generic archiving support elsewhere in pants: https://github.com/pantsbuild/pants/blob/0aa45f8b01be4eaac01f7e2f084d0a25d3963cd9/src/python/pants/fs/archive.py#L148

Reply:
Nice - but it doesn't look like that archiver can archive just one file in a directory. It expects to walk over an entire directory. Plus, it would be strictly more lines of code to use our archiver than this shutil... Maybe, in fact, we should get rid of ours and use the standard one.